homeassistant/components/jvc_projector/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The jvc_projector integration."""
+
+from __future__ import annotations
+
+from jvcprojector import JvcProjector, JvcProjectorAuthError, JvcProjectorConnectError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import JvcProjectorDataUpdateCoordinator
+
+PLATFORMS = [Platform.REMOTE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up integration from a config entry."""
+    device = JvcProjector(
+        host=entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        password=entry.data[CONF_PASSWORD],
+    )
+
+    try:
+        await device.connect(True)
+    except JvcProjectorConnectError as err:
+        await device.disconnect()
+        raise ConfigEntryNotReady(
+            f"Unable to connect to {entry.data[CONF_HOST]}"
+        ) from err
+    except JvcProjectorAuthError as err:
+        await device.disconnect()
+        raise ConfigEntryAuthFailed("Password authentication failed") from err
```

Comment:
The coordinator does the same checks? Why not combine it in the coordinator?

Reply:
These are handling two different function calls. This block is catching errors from the initial device connect early, which can include an invalid hostname. The coordinator is checking for errors doing the scheduled get_state() call.

Reply:
Also __init__.py raises `ConfigEntryNotReady` while the coordinator raises `UpdateFailed`.

Reply:
This is all taken care of by `await coordinator.async_config_entry_first_refresh()` below so agree that this should be removed or it's checking same thing twice.

Reply:
Missed this. Could you look at that in a follow up pull?